lldb/tools/lldb-dap/lldb-dap.cpp

```diff
@@ -3111,17 +3115,20 @@ void request_stackTrace(const llvm::json::Object &request) {
     // This will always return an invalid thread when
     // libBacktraceRecording.dylib is not loaded or if there is no extended
     // backtrace.
-    lldb::SBThread queue_backtrace_thread =
-        thread.GetExtendedBacktraceThread("libdispatch");
+    lldb::SBThread queue_backtrace_thread;
+    if (g_dap.enable_display_extended_backtrace)
+      queue_backtrace_thread = thread.GetExtendedBacktraceThread("libdispatch");
     if (queue_backtrace_thread.IsValid()) {
       // One extra frame as a label to mark the enqueued thread.
       totalFrames += queue_backtrace_thread.GetNumFrames() + 1;
     }
 
     // This will always return an invalid thread when there is no exception in
     // the current thread.
-    lldb::SBThread exception_backtrace_thread =
-        thread.GetCurrentExceptionBacktrace();
+    lldb::SBThread exception_backtrace_thread;
+    if (g_dap.enable_display_extended_backtrace)
+      exception_backtrace_thread = thread.GetCurrentExceptionBacktrace();
```

Comment:
Maybe we should remove this from the stack trace and add this info to the `exceptionInfo` request (https://microsoft.github.io/debug-adapter-protocol/specification#Requests_ExceptionInfo) there is a field for the stack trace in the exception details (https://microsoft.github.io/debug-adapter-protocol/specification#Types_ExceptionDetails).

Reply:
This seems like a better option. What do you think @jeffreytan81 ?

Reply:
This sounds good to me.

@ashgti, feel free to create a PR on top of this one to remove the hot path out of regular stepping code path.

Reply:
Given that you are on this, why don't you give it a try, @jeffreytan81 ?

Reply:
I don't have time to work on this. My goal is to quickly fix the performance regression and unblock the business.

Frankly, the original PR should have been reverted initially, as it introduced the regression.

The reasonable engineering process should involve landing this PR to disable the performance regression. Then, the author of the original PR can fix it in the desired way.

Reply:
🤷 
This been broken for a year, and now you are "rushing". You could as well do a proper fix that works for your company and for everyone else, but well, probably the open source community is not a priority for you.

Reply:
I do not understand why you are judging people. Everyone has their own priority and focus. I am doing a favor here to fix a serious performance regression for the community. Please be respectful in communication.

Reply:
I can take a look at moving this into the `exceptionInfo` request.

Reply:
Expression evaluation can also take longer than 1 second depending on the amount of total debug info and if there are accelerator tables.